sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson

```diff
@@ -1167,5 +1167,18 @@
       stage: V2
       tests: []
     }
+
+    {
+      name: rom_e2e_weak_straps
+      desc: '''Verify that ROM can differentiate strong pull-ups/-downs from weak pull-ups/-downs.
+
+      - For `strap` in {`bootstrap`, `rma_entry`}
+        - Apply all combinations of `strap` where pin values are the same but at least one pin uses weak pull-down.
+        - Verify that ROM boots as usual.
```

Comment:
Just for my education, why do we expect ROM to boot as usual if the pull-down is weak rather than the expected strong? (I'm still not clear what exactly a "strong pull-down" setting means -- the pinmux documentation doesn't really explain.)

Reply:
Sure, we use the level along with the drive strength to encode 2 bits of information to each pin. Since all ROM encodings use strong pull-up/-downs, any weak pull-up/-down should be ignored by ROM because they will be used for other purposes. The ROM should respond to only its specific strap configurations. I highlighted weak vs strong here because checking/testing the levels of the pins is trivial.

Reply:
Ah, that makes sense, thanks for the clarification!

Reply:
i'm sorry we should really document this in the open somewhere. But +1 to everything @alphan is saying!